x-pack/lib/filters/geoip/download_manager.rb

```diff
@@ -67,19 +70,23 @@ def download_database(server_db)
     end
   end
 
+  # extract COPYRIGHT.txt, LICENSE.txt and GeoLite2-{ASN,City}.mmdb from .tgz to temp directory
   def unzip(zip_path)
-    database_path = zip_path[0...-3]
-    Zlib::GzipReader.open(zip_path) do |gz|
-      ::File.open(database_path, "wb") do |f|
-        f.print gz.read
-      end
-    end
-    database_path
+    new_database_path = zip_path[0...-(GZ_EXT.length)] + DB_EXT
+    temp_dir = Stud::Temporary.pathname
+
+    LogStash::Util::Tar.extract(zip_path, temp_dir)
+    logger.debug("extract database to ", :path => temp_dir)
+
+    FileUtils.cp("#{temp_dir}/#{database_name}", new_database_path)
+    FileUtils.cp_r(Dir.glob("#{temp_dir}/{COPYRIGHT,LICENSE}.txt"), @vendor_path)
+
+    new_database_path
   end
 
   # Make sure the path has usable database
   def assert_database!(database_path)
-    raise "failed to load database #{database_path}" unless org.logstash.filters.GeoIPFilter.database_valid?(database_path)
+    raise "failed to load database #{database_path}" unless org.logstash.filters.geoip.GeoIPFilter.database_valid?(database_path)
```

Comment:
`logstash-filter-geoip` rename the package to `org.logstash.filters.geoip` and is waiting to merge that cause the failure of CI. Once [the plugin](https://github.com/logstash-plugins/logstash-filter-geoip/pull/179) is merged, CI should be green